# Iris Flower Classification Using Amazon SageMaker Studio

## Project Overview

This beginner machine-learning project uses Amazon SageMaker Studio and Python to classify iris flowers as Setosa, Versicolor, or Virginica.

The model uses four flower measurements:

- Sepal length
- Sepal width
- Petal length
- Petal width

A Random Forest classifier is trained and evaluated using accuracy, precision, recall, F1-score, and a confusion matrix.

**Model accuracy: 90%**

In [1]:
from sklearn.datasets import load_iris
import pandas as pd

iris = load_iris()

X = pd.DataFrame(iris.data, columns=iris.feature_names)
y = iris.target

print("Dataset loaded successfully!")
print("Number of flowers:", len(X))
X.head()

Dataset loaded successfully!
Number of flowers: 150


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm)
0,5.1,3.5,1.4,0.2
1,4.9,3.0,1.4,0.2
2,4.7,3.2,1.3,0.2
3,4.6,3.1,1.5,0.2
4,5.0,3.6,1.4,0.2


## 2. Split the Dataset and Train the Model

The dataset is divided into training and testing portions.

A Random Forest model is trained to predict the iris flower species using its measurements.

In [2]:
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.20,
    random_state=42,
    stratify=y
)

model = RandomForestClassifier(
    n_estimators=100,
    random_state=42
)

model.fit(X_train, y_train)

print("Model training completed!")
print("Training samples:", len(X_train))
print("Testing samples:", len(X_test))

Model training completed!
Training samples: 120
Testing samples: 30


In [3]:
from sklearn.metrics import accuracy_score, classification_report

# Ask the model to predict the test flowers
y_pred = model.predict(X_test)

# Compare predictions with the correct answers
accuracy = accuracy_score(y_test, y_pred)

print(f"Model accuracy: {accuracy:.2%}")
print("\nClassification Report:")
print(
    classification_report(
        y_test,
        y_pred,
        target_names=iris.target_names
    )
)

Model accuracy: 90.00%

Classification Report:
              precision    recall  f1-score   support

      setosa       1.00      1.00      1.00        10
  versicolor       0.82      0.90      0.86        10
   virginica       0.89      0.80      0.84        10

    accuracy                           0.90        30
   macro avg       0.90      0.90      0.90        30
weighted avg       0.90      0.90      0.90        30



import matplotlib.pyplot as plt
from sklearn.metrics import ConfusionMatrixDisplay

ConfusionMatrixDisplay.from_predictions(
    y_test,
    y_pred,
    display_labels=iris.target_names,
    cmap="Blues"
)

plt.title("Iris Classification Confusion Matrix")
plt.tight_layout()
plt.show()

In [5]:
# Measurements: sepal length, sepal width, petal length, petal width
new_flower = pd.DataFrame(
    [[5.1, 3.5, 1.4, 0.2]],
    columns=iris.feature_names
)

# Predict the flower species
predicted_number = model.predict(new_flower)[0]
predicted_species = iris.target_names[predicted_number]

# Find the model's confidence
probabilities = model.predict_proba(new_flower)[0]
confidence = probabilities[predicted_number]

print("New flower measurements:")
display(new_flower)

print("Predicted species:", predicted_species)
print(f"Prediction confidence: {confidence:.2%}")

New flower measurements:


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm)
0,5.1,3.5,1.4,0.2


Predicted species: setosa
Prediction confidence: 100.00%


## 6. Conclusion

This project trained a Random Forest classifier using the Iris dataset.

### Results

- The dataset contained 150 flower samples.
- 120 samples were used for training.
- 30 samples were used for testing.
- The model achieved 90% testing accuracy.
- It correctly classified all Setosa flowers.
- Most errors occurred between Versicolor and Virginica because their measurements can overlap.
- The model successfully predicted a new flower as Setosa with 100% confidence.

### What I Learned

Through this project, I practiced loading data, splitting data into training and testing sets, training a machine-learning model, measuring performance, interpreting a confusion matrix, and making a new prediction using Amazon SageMaker Studio.